### PR TITLE
Basic backup and juju-restore bash tests

### PR DESCRIPTION
**Pull request by benhoyt**
We don't have any CI tests for the new stand-alone [`juju-restore`](https://github.com/juju/juju-restore) tool, and our current backup/restore tests are Python acceptance tests. This PR adds a basic "sanity check" test for `juju create-backup`, as well as a basic end-to-end test of `juju-restore`.

### CODE REVIEW

**Comment by tlm** [comment]
lgtm

**Review by tlm — APPROVED** [review]

**Review by wallyworld — APPROVED** [review]
Thanks, more tests good

File: tests/suites/backup_restore/backup_restore.sh [hunk 1]
```diff
@@ -0,0 +1,71 @@
+run_basic_backup_create() {
+    set -e  # TODO benhoyt: remove once "set +e" issue is fixed
+    echo
+
+    juju switch controller
```
Comment (SimonRichardson) [inline tests/suites/backup_restore/backup_restore.sh:5]: I'm assuming that you can only backup from a controller? If so can we add a note about why you're not using ensure here..?

**Review by SimonRichardson — CHANGES_REQUESTED** [review]

File: tests/suites/backup_restore/backup_restore.sh (hunk 1 above)
Comment (benhoyt) [reply tests/suites/backup_restore/backup_restore.sh:4]: Yes - clarified with comment. Not sure why I would use "ensure" here?

File: tests/suites/backup_restore/backup_restore.sh [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    juju add-unit ubuntu-lite
+    juju status --format json | jq '.machines | length' | check 2
+
+    echo "Restore the backup"
+    juju switch controller
+    juju scp "${TEST_DIR}/juju-restore" 0:
+    juju scp "${TEST_DIR}/basic_backup.tar.gz" 0:
+    juju ssh 0 ./juju-restore --yes basic_backup.tar.gz
+
+    echo "Ensure there's only one machine (state before the backup)"
```
[the remark this answers is not in the document]
Comment (benhoyt) [reply tests/suites/backup_restore/backup_restore.sh:50]: Added (lxd-only)

File: tests/suites/backup_restore/backup_restore.sh (hunk 1 above)
Comment (benhoyt) [reply tests/suites/backup_restore/backup_restore.sh:8]: Ah, I see what you mean by "ensure" now -- fixed, thanks.

**Comment by benhoyt** [comment]
!!build!!

**Review by hpidcock — APPROVED** [review]

**Comment by benhoyt** [comment]
!!build!!

**Comment by benhoyt** [comment]
$$merge$$

**Review by SimonRichardson — APPROVED** [review]

**Comment by SimonRichardson** [comment]
$$merge$$

**Pull request merged**